src/renderers/dom/shared/__tests__/ReactDOMServerIntegration-test.js

```diff
@@ -455,6 +455,22 @@ describe('ReactDOMServerIntegration', () => {
         expect(parent.childNodes[1].tagName).toBe('DIV');
         expect(parent.childNodes[2].tagName).toBe('DIV');
       });
+
+      itRenders('emptyish values', async render => {
+        let e = await render(0);
+        expect(e.nodeType).toBe(TEXT_NODE_TYPE);
+        expect(e.nodeValue).toMatch('0');
+
+        // TODO: This one is broken because client renders a node
+        // but server returns empty HTML.
+        // expect(await render('')).toBe(null);
```

Comment:
In some test I've instead tested that `parent.textContent === ''` since that's what really matters. That there's no significant text content. More resilient to algorithm changes.

You could do:

```js
expect((await render(<div>{''}</div>)).textContent).toBe('');
```